maestro-xcuitest-driver/src/main/kotlin/xcuitest/installer/LocalXCTestInstaller.kt

```diff
@@ -26,29 +26,36 @@ class LocalXCTestInstaller(
 
     private var xcTestProcess: Process? = null
 
-    override fun killAndUninstall() {
+    override fun uninstall() {
         if (useXcodeTestRunner) {
             return
         }
 
+        stop()
+
+        logger.info("[Start] Uninstalling the XCUITest runner app")
+        XCRunnerSimctl.uninstall(UI_TEST_RUNNER_APP_BUNDLE_ID)
+        logger.info("[Done] Uninstalling the XCUITest runner app")
+    }
+
+    private fun stop() {
         if (xcTestProcess?.isAlive == true) {
             logger.info("[Start] Killing the started XCUITest")
             xcTestProcess?.destroy()
             logger.info("[Done] Killing the started XCUITest")
         }
-        logger.info("[Start] Uninstalling the XCUITest runner app")
-        XCRunnerSimctl.uninstall(UI_TEST_RUNNER_APP_BUNDLE_ID)
-        logger.info("[Done] Uninstalling the XCUITest runner app")
     }
 
-    override fun setup(): Boolean {
+    override fun start(): Boolean {
         if (useXcodeTestRunner) {
             return ensureOpen()
         }
 
+        stop()
```

Comment:
Why not uninstall here and only stop?

Reply:
Because it would take time without any benefit.

Reply:
Got it! Yup if the app is not going to be present here I think we are ok.